library/std/src/sync/barrier.rs

```diff
@@ -130,11 +130,8 @@ impl Barrier {
         let local_gen = lock.generation_id;
         lock.count += 1;
         if lock.count < self.num_threads {
-            // We need a while loop to guard against spurious wakeups.
-            // https://en.wikipedia.org/wiki/Spurious_wakeup
-            while local_gen == lock.generation_id {
-                lock = self.cvar.wait(lock).unwrap();
-            }
+            let _guard =
```

Comment:
It's a small thing, but why not still reassign to `lock` so we only have one guard? I know that conditionally only one of them will be occupied, but there's no reason to make the compiler prove that...

Reply:
the answer is in the [build error](https://github.com/rust-lang/rust/actions/runs/5734915823/job/15541863600) caused by my previous revision that did exactly what you're asking for :) TL;DR;
```
  error: value assigned to `lock` is never read
     --> library/std/src/sync/barrier.rs:133:13
      |
  133 |             lock = self.cvar.wait_while(lock, |state| local_gen == state.generation_id).unwrap();
      |             ^^^^
      |
      = help: maybe it is overwritten before being read?
      = note: `-D unused-assignments` implied by `-D warnings`
  
  error: could not compile `std` (lib) due to previous error
```

Reply:
Ugh, that's annoying, but ok.